Azure/azure-cli
# [RDBMS] `az mysql flexible-server upgrade`: Add major version upgrade for MySQL flexible server

**Related command**
`az mysql flexible-server upgrade`

**Description**
This PR enables major version upgrades for MySQL flexible server.

**Testing Guide**
Example of how to create a server with version 5.7 and update it to version 8:
- `az mysql flexible-server create -g testgroup -n testsvr --public-access none --tier GeneralPurpose --sku-name Standard_D2ds_v4 --location eastus --version 5.7`
- `az mysql flexible-server parameter set -g testgroup -s testsvr -n sql_mode -v "ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO"`
- `az mysql flexible-server upgrade -g testgroup -n testsvr --version 8`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
